Link C++ coverage to python tests

parijatm opened the pull request.

parijatm commented:
Getting C++ coverage from python tests already works @stevenmburns (See test-align artifacts). The coverage will change depending upon the test suite. Should I be doing this for test-integration as well? I was thinking we should just do a better job of getting coverage from test-align.

parijatm commented:
@Lastdayends @854768750 I removed the Dockerfiles & run*docker*.sh files in PlaceRouteHierFlow as test-PlaceRouteHierFlow is using the converged align image. Please confirm they weren't being used. 

I also see a large number of files & folders (most testcase_* folders, run.csh, run_drc.sh among others) which don't really seem to be used. Can you please clean up PlaceRouteHierFlow a little bit? I am trying to get the docker image size down a bit.

854768750 commented:
> @Lastdayends @854768750 I removed the Dockerfiles & run_docker_.sh files in PlaceRouteHierFlow as test-PlaceRouteHierFlow is using the converged align image. Please confirm they weren't being used.
> 
> I also see a large number of files & folders (most testcase_* folders, run.csh, run_drc.sh among others) which don't really seem to be used. Can you please clean up PlaceRouteHierFlow a little bit? I am trying to get the docker image size down a bit.

I do not use these files. I am not sure @Lastdayends need the files. Please check below.
Here are some not longer used files:
EM.txt
gridresult.txt
InputCurrent_initial.txt
IR_drop_1C.txt
IR_drop_hspice.txt
IR_drop.txt
Power_Grid_Conf.txt
run_drc.sh
run.csh

I think we can save a copy of them and add them to .gitignore

parijatm commented:
> I think we can save a copy of them and add them to .gitignore

Save a copy of them in the repo itself or do you want to make a local copy in your workspace? I would like to move them to 'dev' if they are simply backups & no longer needed in our flow. Updating .gitignore is only needed if this is a generated file that you don't want to add to the repo.

854768750 commented:
> > I think we can save a copy of them and add them to .gitignore
> 
> Save a copy of them in the repo itself or do you want to make a local copy in your workspace? I would like to move them to 'dev' if they are simply backups & no longer needed in our flow. Updating .gitignore is only needed if this is a generated file that you don't want to add to the repo.

I mean a local copy. They are not required by the flow. I think these are generated debug log files.

stevenmburns reviewed (approved):
Looks good. Check it in when you are ready.

stevenmburns commented:
@parijatm The lcov stuff looks really nice. Thanks for getting rid of the coverage in the library code.
Doing coverage is probably a good reason to keep the "-g" flag.

parijatm commented:
> @parijatm The lcov stuff looks really nice. Thanks for getting rid of the coverage in the library code.
> Doing coverage is probably a good reason to keep the "-g" flag.

Coverage actually works fine without the -g as well (using --coverage). Since I don't think anyone is going to be doing debug unless in an interactive session, I split up make into two:
`make all`
Uses flags:
OPTIMFLAGS?=-O3 -s -w
`make debug`
DEBUGFLAGS?=-O3 -g -Wall # Consider using -Og instead of -O3

The default target is `all` which means we are using -s instead of -g (for reduced docker image size). 

I also noticed we were using both -Wall and -w simultaneously so I split them up as well assuming developers will want to see the warnings and users will not.

@Lastdayends @854768750 you should be aware of this change as well.
